### PR TITLE
V2 instructions with stake entry kind

### PR DESCRIPTION
**Pull request by gchatz22**

### CODE REVIEW

File: programs/cardinal-stake-pool/src/instructions/init_entry.rs [hunk 1]
```diff
@@ -38,6 +38,8 @@ pub fn handler(ctx: Context<InitEntryCtx>, _user: Pubkey) -> Result<()> {
     stake_entry.pool = ctx.accounts.stake_pool.key();
     stake_entry.original_mint = ctx.accounts.original_mint.key();
     stake_entry.amount = 0;
+    // every new stake en
```
Comment (jpbogle) [inline programs/cardinal-stake-pool/src/instructions/init_entry.rs:41]: comment half done haha

File: tests/rewards-multiplier-test.spec.ts [hunk 2]
```diff
@@ -265,31 +266,6 @@ describe("Stake and claim rewards", () => {
...
-      provider.connection,
-      rewardDistributorId
-    );
-    const a = await checkMint.getAccountInfo(userRewardMintTokenAccount);
-    console.log("user reward mint token acount", a.amount.toNumber());
-    console.log(
-      "rewardsIssued",
-      rewardDistributorData.parsed.rewardsIssued.toNumber()
-    );
-
```
Comment (jpbogle) [inline tests/rewards-multiplier-test.spec.ts:292]: we dont want this test anymore? this check still seems "correct" from the tests perspective....

u claim rewards and u check if they got the rewards no

File: programs/cardinal-reward-distributor/src/instructions/claim_rewards_v2.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    #[account(mut, constraint = user_reward_mint_token_account.mint == reward_distributor.reward_mint && user_reward_mint_token_account.owner == stake_entry.last_staker @ ErrorCode::InvalidUserRewardMintTokenAccount)]
+    user_reward_mint_token_account: Box<Account<'info, TokenAccount>>,
+
+    /// CHECK: This is not dangerous because we don't read or write from this account
+    #[account(mut, constraint = assert_reward_manager(&reward_manager.key()))]
+    reward_manager: UncheckedAccount<'info>,
+
+    #[account(mut)]
+    user: Signer<'info>,
```
Comment (jpbogle) [inline programs/cardinal-reward-distributor/src/instructions/claim_rewards_v2.rs:37]: should this not be checking last_staker?

File: tests/rewards-multiplier-test.spec.ts (hunk 2 above)
Comment (gchatz22) [reply tests/rewards-multiplier-test.spec.ts:292]: there was no check happening, assumed it was logging code and just removed automatically. will add proper check

File: programs/cardinal-reward-distributor/src/instructions/claim_rewards_v2.rs (hunk 3 above)
Comment (gchatz22) [reply programs/cardinal-reward-distributor/src/instructions/claim_rewards_v2.rs:37]: `user` acts as payer here, the check for `last_staker` happens on the user token account

File: programs/cardinal-stake-pool/src/instructions/init_entry.rs (hunk 1 above)
Comment (gchatz22) [reply programs/cardinal-stake-pool/src/instructions/init_entry.rs:41]: word

**Comment by gchatz22** [comment]
closing as already addressed

**Pull request closed without merging**